stdlib/src/builtin/format_int.mojo

```diff
@@ -28,7 +28,37 @@ alias _DEFAULT_DIGIT_CHARS = "0123456789abcdefghijklmnopqrstuvwxyz"
 # ===----------------------------------------------------------------------===#
 
 
-fn bin(num: Scalar, /, *, prefix: StaticString = "0b") -> String:
+fn bin[
+    WriterType: Writer, //
+](value: Scalar, /, *, inout writer: WriterType, prefix: StaticString = "0b"):
```

Comment:
Nit: make `writer` the last arg so it's consistent with `print`.

Reply:
hmm I might like that as well, hang on.

Reply:
Mojo doesn't let us put a required keyword-only argument after an optional keyword-only argument